.github/workflows/main.yml

```diff
@@ -84,3 +84,6 @@ jobs:
 
       - name: Run pytest
         run: poetry run python -m pytest -p no:sugar -q tests/
+
+      - name: Run mypy
+        run: poetry run mypy src
```

Comment:
Do we need this run for every platform/python combination?

Reply:
No, I don't think so, it is enough with the highest version of python and in ubuntu. mypy does not change in other environments.

Reply:
> Do we need this run for every platform/python combination?

I think, we should run it on every platform. #5135 proved that mypy can fail on Windows even if it succeeds on Linux.

Reply:
maybe include `tests` as well?

Reply:
Running against all platform/python combination seems to me a bit overkill. I would prefer that `mypy` gets its own job and is running against all platform but only one for one specified python. (But maybe spinning up these jobs have more overhead, than running within the "tests" job? :thinking: )

Furthermore we should drop `mypy` from the `pre-commit` checks.

Reply:
I put it all in the same job because indeed the cost of running mypy is surely dwarfed by the cost of installing poetry and running poetry install.  I kind of agree that it would be cleaner to have it separate; but also I don't care very much.

Similarly on multiple platform/python combinations - different platforms and python versions can give different results so it's not crazy to run against all of them.  On the other hand I don't suppose we'd lose very much by just picking one; I don't really care about that detail either.

I mostly intend this specific MR as a demonstration that the pre-commit approach is missing stuff, and that we should do better.  Feel free either to merge this as it is, or a rearranged version as any of you prefer.